### PR TITLE
Editorial: Extend GetOption to support a *"number"* type

### PR DESCRIPTION
**Pull request by gibson042**
Ref https://github.com/tc39/proposal-temporal/pull/2243

### CODE REVIEW

**Review by ryzokuken — APPROVED** [review]
LGTM overall. I suppose it's editorial since it changes an AO that isn't exposed directly but it does change the behavior around `type: <something wrong>`, although I guess it's fair to assume that this case just wouldn't happen.

Also, not sure how `~required~` works?

**Comment by gibson042** [comment]
@ryzokuken There is one use of GetOption in ECMA-402 that _could_ have used `~required~`, except that it throws a TypeError rather than a RangeError when the required property is undefined: the Intl.DisplayNames constructor reading a "type" option (https://tc39.es/ecma402/#sec-Intl.DisplayNames ).

File: spec/negotiation.html [hunk 1]
```diff
@@ -327,21 +327,32 @@ <h1>CoerceOptionsToObject ( _options_ )</h1>
...
+          1. If _default_ is ~required~, throw a *RangeError* exception.
+          1. Return _default_.
         1. If _type_ is *"boolean"*, then
           1. Set _value_ to ! ToBoolean(_value_).
-        1. If _type_ is *"string"*, then
+        1. If _type_ is *"number"*, then
+          1. Set _value_ to ? ToNumber(_value_).
+          1. If _value_ is *NaN*, throw a *RangeError* exception.
+        1. Else,
+          1. Assert: _type_ is *"string"*.
```
Comment (ljharb) [inline spec/negotiation.html:355]: Why are these strings and not spec enums?

Comment (gibson042) [reply spec/negotiation.html:355]: I was just about to open a discussion on that topic. ECMA-402 currently uses `*"boolean"*`/`*"number"*`/`*"string"*` strings, Temporal currently uses direct `Boolean`/`String`/`Number` [Type](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-ecmascript-data-types-and-values) references, and the more conventional approach would be a spec enum like `~boolean~`/`~number~`/`~string~`. The string approach is obviously dispreferred, but how do we feel about direct Type vs. spec enum?

**Pull request merged**